main/main.ino

```diff
@@ -1037,106 +1049,105 @@ void loop() {
 
 #if defined(ESP8266) || defined(ESP32)
   if (WiFi.status() == WL_CONNECTED) {
+    ArduinoOTA.handle();
 #else
   if ((Ethernet.hardwareStatus() != EthernetW5100 && Ethernet.linkStatus() == LinkON) || (Ethernet.hardwareStatus() == EthernetW5100)) { //we are able to detect disconnection only on w5200 and w5500
 #endif
     failure_number_ntwk = 0;
-    if (client.connected()) {
+    if (!ProcessLock) { // If no blocking process we can handle MQTT processing
```

Comment:
can we instead do something like?
```c++
if (ProcessLock) return;
```

Reply:
yes we can

Reply:
After looking into the [code](https://github.com/esp8266/Arduino/blob/master/libraries/ArduinoOTA/ArduinoOTA.cpp)  `ProcessLock` should never be `true` here, because when `ArduinoOTA.handle()` returns there is no update in progress. Either there was no update or it failed. When the update was successful the board is restarted and `ArduinoOTA.handle()` should not have returned.

Reply:
ok I will remove this check

Reply:
It was added due to the fact that we were seeing MQTT messages during the OTA process:
https://community.openmqttgateway.com/t/ota-upload-on-esp32/1013/5?u=1technophile

Reply:
the mqtt connect comes after the `Error[4]: E: End Failed` so it's after the OTA failed. The OTA kills all network connections before the update, this includes the mqtt connection. After a failed update the mqtt client reconnects. You don't see that on an successful update because in that case the esp is restarted.

Reply:
Thanks for the analysis!